applications/GeoMechanicsApplication/custom_strategies/schemes/newmark_quasistatic_U_Pw_scheme.hpp

```diff
@@ -186,32 +154,27 @@ class NewmarkQuasistaticUPwScheme : public GeoMechanicsTimeIntegrationScheme<TSp
     {
         KRATOS_TRY
 
-        //Update Acceleration, Velocity and DtPressure
+        // Update Acceleration, Velocity and DtPressure
         block_for_each(rModelPart.Nodes(), [&](Node& rNode) {
-            noalias(rNode.FastGetSolutionStepValue(ACCELERATION)) =  ((  rNode.FastGetSolutionStepValue(DISPLACEMENT)
-                                                                      - rNode.FastGetSolutionStepValue(DISPLACEMENT, 1))
-                                                                   - this->GetDeltaTime() * rNode.FastGetSolutionStepValue(VELOCITY, 1)
-                                                                   - (0.5-mBeta) * this->GetDeltaTime() * this->GetDeltaTime()
-                                                                   * rNode.FastGetSolutionStepValue(ACCELERATION, 1) )
-                                                                   / (mBeta*this->GetDeltaTime()*this->GetDeltaTime());
-
-            noalias(rNode.FastGetSolutionStepValue(VELOCITY)) =  rNode.FastGetSolutionStepValue(VELOCITY, 1)
-                                                                + (1.0-mGamma)* this->GetDeltaTime()
-                                                                * rNode.FastGetSolutionStepValue(ACCELERATION, 1)
-                                                                + mGamma * this->GetDeltaTime()
-                                                                * rNode.FastGetSolutionStepValue(ACCELERATION);
-
-            const double DeltaPressure =  rNode.FastGetSolutionStepValue(WATER_PRESSURE)
-                                        - rNode.FastGetSolutionStepValue(WATER_PRESSURE, 1);
-
-            rNode.FastGetSolutionStepValue(DT_WATER_PRESSURE) = ( DeltaPressure
-                                                                 - (1.0-mTheta) * this->GetDeltaTime()
-                                                                  * rNode.FastGetSolutionStepValue(DT_WATER_PRESSURE, 1))
-                                                                / (mTheta* this->GetDeltaTime());
+            noalias(rNode.FastGetSolutionStepValue(ACCELERATION)) =
+                ((rNode.FastGetSolutionStepValue(DISPLACEMENT) -
+                  rNode.FastGetSolutionStepValue(DISPLACEMENT, 1)) -
+                 this->GetDeltaTime() * rNode.FastGetSolutionStepValue(VELOCITY, 1) -
+                 (0.5 - mBeta) * this->GetDeltaTime() * this->GetDeltaTime() *
+                     rNode.FastGetSolutionStepValue(ACCELERATION, 1)) /
+                (mBeta * this->GetDeltaTime() * this->GetDeltaTime());
+
+            noalias(rNode.FastGetSolutionStepValue(VELOCITY)) =
+                rNode.FastGetSolutionStepValue(VELOCITY, 1) +
+                (1.0 - mGamma) * this->GetDeltaTime() *
+                    rNode.FastGetSolutionStepValue(ACCELERATION, 1) +
+                mGamma * this->GetDeltaTime() * rNode.FastGetSolutionStepValue(ACCELERATION);
```

Comment:
Could be factored out as UpdateVectorFirstTimeDerivative and UpdateVectorSecondTimeDerivative.